### ISSUE TITLE
Add support for Node 12

### ISSUE THREAD

**Issue by ColbyTresness** [microsoft/vscode-azurefunctions#1713]
The functions service will soon allow hosting Node 12 applications, and we should add support in the tooling for: 
- version selection on creation (similar to what we do for Python)
- validation on publishing (local version matches Azure version)
- Once platform is ready, default should be Node 12 and Functions V3

**Comment by ColbyTresness** [comment]
Also note that when a user upgrades to V3 we should ensure that their node versions are compatible with that runtime version

**Comment by johnpapa** [comment]
I was just about to open an issue for this. I received

```
[error] Incompatible Node.js version. The version you are using is v12.13.1, but the runtime requires an LTS-covered major version (ex: 8.11.1 or 10.14.1). LTS-covered versions have an even major version number (8.x, 10.x, etc.) as per https://github.com/nodejs/Release#release-plan. For deployed code, change WEBSITE_NODE_DEFAULT_VERSION in App Settings. Locally, install or switch to a supported node version (make sure to quit and restart your code editor to pick up the changes).
```

Seems the tools need to be updated for Node 12 as an LTS. RIght?

cc @fiveisprime

**Comment by ColbyTresness** [comment]
Yeah in the interim while we don't support Node 12, we should update this string

**Comment by johnpapa** [comment]
Thanks for the reply @ColbyTresness 

This may cause some friction when we start publishing the 25 days of Serverless in December, @simonaco (just as an FYI).

Anyone trying with Node 12 who tries these may run into it. If it won't be supported by Dec 1, then we may want to add instructions to each challenge to mention using Node 10.

**Comment by EricJizbaMSFT** [comment]
@ColbyTresness we already support multiple versions of Node (8 and 10). It sounds like all we need to do is add 12 to that list, right? Your original comment includes some stuff that I believe is already done.

In regards to the message @johnpapa mentioned, that error is coming from the func cli, not our extension. I believe updates to that message are being tracked here: https://github.com/Azure/azure-functions-nodejs-worker/issues/200

**Comment by ColbyTresness** [comment]
Basically, yes, but I'd wait a bit until the platform is completely ready. Forwarding you an email where I explain the entire situation.
